### PR TITLE
update DfmVideoBlockRule.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Dfm/Rules/DfmVideoBlockRule.cs [hunk 1]
```diff
@@ -7,20 +7,52 @@ namespace Microsoft.DocAsCode.Dfm
...
+                ).RepeatAtLeast(1)
+            ).ToGroup("link") +
+            Matcher.WhiteSpacesOrEmpty + ']' + Matcher.WhiteSpacesOrEmpty +
+            (Matcher.NewLine.RepeatAtLeast(1) | Matcher.EndOfString);
+
         private static readonly Regex _videoRegex = new Regex(@"^ *\[\!Video +(?<link>https?\:\/\/.+?) *\] *(\n|$)", RegexOptions.Compiled | RegexOptions.IgnoreCase, TimeSpan.FromSeconds(10));
+
         public virtual string Name => "DfmVideoBlock";
+
+        [Obsolete("Please use NewLineMatcher.")]
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Dfm/Rules/DfmVideoBlockRule.cs]: >NewLineMatcher [](start = 30, length = 14)

VideoMatcher

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Dfm/Rules/DfmVideoBlockRule.cs [hunk 2]
```diff
@@ -7,20 +7,52 @@ namespace Microsoft.DocAsCode.Dfm
     using System.Text.RegularExpressions;
 
     using Microsoft.DocAsCode.MarkdownLite;
+    using Microsoft.DocAsCode.MarkdownLite.Matchers;
 
     public class DfmVideoBlockRule : IMarkdownRule
     {
+        private static readonly Matcher _VideoMatcher =
+            Matcher.WhiteSpacesOrEmpty + Matcher.CaseInsensitiveString("[!video") + Matcher.WhiteSpaces +
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Dfm/Rules/DfmVideoBlockRule.cs]: >WhiteSpaces [](start = 92, length = 11)

WhiteSpacesOrEmpty？

Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Dfm/Rules/DfmVideoBlockRule.cs]: in regex is " +".

**Pull request merged**